HLTrigger/HLTcore/interface/HLTPrescaleProvider.h

```diff
@@ -58,25 +59,53 @@ class HLTPrescaleProvider {
   // negative == error
 
   /// combining the two methods above
-  unsigned int prescaleValue(const edm::Event& iEvent, const edm::EventSetup& iSetup, const std::string& trigger);
-
+  template<typename T = unsigned int>
+  T prescaleValue(const edm::Event& iEvent, const edm::EventSetup& iSetup, const std::string& trigger){
+    const int set(prescaleSet(iEvent, iSetup));
+    return set < 0 ? -1 : 
+      hltConfigProvider_.prescaleValue<T>(static_cast<unsigned int>(set), trigger);
+  }
+  
   /// Combined L1T (pair.first) and HLT (pair.second) prescales per HLT path
-  std::pair<int, int> prescaleValues(const edm::Event& iEvent,
-                                     const edm::EventSetup& iSetup,
-                                     const std::string& trigger);
+  template<typename T = int>
+  std::pair<T, T> prescaleValues(const edm::Event& iEvent,
```

Comment:
Should one allow T1,T2 so that the types for L1T and HLT could be different?

Reply:
so, this is what I was wondering, its a good point. I initially went avoided this road because of specialisation nightmares but then I realised I could of course just specialise the HLT and L1 functions separately. 

So unless it gets difficult to do so, I'll add that feature

Reply:
done